department-of-veterans-affairs/caseflow
# Add cavc privacy act admin action

Bumps [CASEFLOW-107]

### Description
Allows cavc users to add translation admin actions.

### Acceptance Criteria
- [ ] Task action "Send to privacy team" is added to SendCavcRemandProcessedLetterTask
- [ ] Provide Instructions and context for this action appears on the submit admin actions page
- [ ] Creates a PrivacyActTask assigned to the PrivacyTeam as a child of the distribution task

### Testing Plan
1. Log in as the CAVC team admin and assign any open `SendCavcRemandProcessedLetterTask` to yourself
1. Click the task action dropdown and confirm new "send to privacy team" action
1. Confirm you can add instructions to this action
1. Confirm clicking submit keeps you on the case details page and creates a privacy act task assigned to the privacy team is created as a child of the distribution task
1. Confirm you still have actions available to reassign or complete
1. Confirm you cannot create another task assigned to the privacy team
```ruby
Appeal 920 (evidence_submission)                       ID   STATUS      ASGN_BY                ASGN_TO                UPDATED_AT
└── RootTask                                           2733 on_hold                            Bva                    2020-11-16 15:55:50 UTC
    └── DistributionTask                               2734 on_hold                            Bva                    2020-11-16 15:55:50 UTC
        ├── CavcTask                                   2735 on_hold                            Bva                    2020-11-16 15:55:50 UTC
        │   └── SendCavcRemandProcessedLetterTask      2736 on_hold                            CavcLitigationSupport  2020-11-17 14:55:18 UTC
        │       └── SendCavcRemandProcessedLetterTask  3365 in_progress CAVC_LIT_SUPPORT_ADMIN CAVC_LIT_SUPPORT_ADMIN 2020-11-17 14:55:22 UTC
        └── PrivacyActTask                             3366 assigned    CAVC_LIT_SUPPORT_ADMIN PrivacyTeam            2020-11-17 14:56:32 UTC
```

### User Facing Changes

 Workflow Step|UI
 ---|---
New task action|<img width="662" alt="Screen Shot 2020-11-17 at 9 55 27 AM" src="https://user-images.githubusercontent.com/45575454/99406661-27b3f580-28bc-11eb-9517-d1658be1bc17.png">
Modal|<img width="462" alt="Screen Shot 2020-11-17 at 9 55 34 AM" src="https://user-images.githubusercontent.com/45575454/99406670-2aaee600-28bc-11eb-9fd7-f9952b09da30.png">
Modal validation|<img width="465" alt="Screen Shot 2020-11-17 at 9 55 38 AM" src="https://user-images.githubusercontent.com/45575454/99406684-2e426d00-28bc-11eb-8cb9-b2564dcc5088.png">
Success & updated task snapshot|<img width="742" alt="Screen Shot 2020-11-17 at 9 56 33 AM" src="https://user-images.githubusercontent.com/45575454/99406696-31d5f400-28bc-11eb-900d-92b3e93c11c7.png">
Error creating duplicate task|<img width="462" alt="Screen Shot 2020-11-17 at 9 56 47 AM" src="https://user-images.githubusercontent.com/45575454/99406712-34384e00-28bc-11eb-8607-87d9ca7e4e4c.png">


### Code Documentation Updates
